lib/ecto/repo/queryable.ex

```diff
@@ -36,6 +36,20 @@ defmodule Ecto.Repo.Queryable do
     one!(repo, adapter, query_for_get(queryable, id), opts)
   end
 
+  def get_by(repo, adapter, queryable, clause, opts) do
+    [key] = Keyword.keys(clause)
+    [value] = Keyword.values(clause)
+
+    one(repo, adapter, query_for_get_by(queryable, key, value), opts)
+  end
+
+  def get_by!(repo, adapter, queryable, clause, opts) do
+    [key] = Keyword.keys(clause)
```

Comment:
Those two lines could have been:

    [{key, value}] = clause

However, we also want this to work:

    Repo.get_by User, email: ..., name: ...

So how could we make it work? :)